include/blackhole/record.hpp

```diff
@@ -15,12 +15,12 @@ using cpp17::string_view;
 
 class record_t {
 public:
+    struct inner_t;
```

Comment:
It's for owned record to be able to initialize this class directly from implementation type. Dark place, agree, but alternative solution involves friendship.